### PR TITLE
support network neighbors

### PR DESCRIPTION
**Pull request by Daniel-GrunbergerCA**
## PR Type:
Enhancement

___
## PR Description:
This PR introduces a new feature to support network neighbors. The main changes include:
- Adding a network tracer to listen to communication.
- Adding a network manager to handle network events.
- Supporting services without selectors.
- Adding tests and debug logs.
- Fixing label selector.
- Handling network events and generating entries for CRD.
- Creating and updating network neighbors CRD based on events.
- Handling services with no selectors.
- Deleting resources when a container is removed.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`pkg/networkmanager/network_manager.go`: This file introduces the NetworkManager struct and its associated methods. The NetworkManager is responsible for handling network events, creating and updating network neighbors CRD based on these events, and deleting resources when a container is removed.
`pkg/containerwatcher/v1/open_test.go`: This file modifies the test setup to accommodate the new NetworkManager parameter in the CreateIGContainerWatcher function.
`pkg/networkmanager/network_manager_test.go`: This file includes tests for the NetworkManager, ensuring its methods function as expected.
`pkg/networkmanager/network_neighbors.go`: This file introduces functions to generate network neighbors CRD and network neighbors entries.
`pkg/containerwatcher/v1/container_watcher.go`: This file modifies the ContainerWatcher struct to include a NetworkManager, and updates the relevant methods to use this manager.
`pkg/networkmanager/network_event.go`: This file introduces the NetworkEvent struct and its associated methods. NetworkEvent represents a network event that occurs in a container.
`pkg/storage/storage_interface.go`: This file updates the StorageClient interface to include methods for creating, getting, patching, and deleting NetworkNeighbors.
`pkg/relevancymanager/v1/relevancy_manager.go`: This file modifies the RelevancyManager to include a NetworkManager, and updates the relevant methods to use this manager.
`pkg/config/config.go`: This file updates the Config struct to include parameters related to the NetworkManager, such as InitialDelay and UpdateDataPeriod.
`main.go`: This file modifies the main function to create a NetworkManager and pass it to the RelevancyManager and ContainerWatcher.
</details>

___
## User Description:
1. Add a network tracer to listen to communication.
2. Add a network manager to handle network events

### CODE REVIEW

File: pkg/networkmanager/network_event.go
Comment (matthyx) [inline pkg/networkmanager/network_event.go:75]: ;-)

**Review by matthyx — COMMENTED** [review]
integration in the node-agent LGTM, unfortunately I don't have enough time to validate the business logic of generating network-policies

File: go.mod
Comment (dwertent) [inline go.mod:16]: Can we release the storage?

Comment (Daniel-GrunbergerCA) [reply go.mod:16]: waiting for vlad approval

File: pkg/storage/v1/storage_nocache.go
Comment (dwertent) [inline pkg/storage/v1/storage_nocache.go:32]:
```suggestion
	kubeconfig := "/home/daniel/.kube/config"
```

File: pkg/relevancymanager/v1/relevancy_manager.go
Comment (dwertent) [inline pkg/relevancymanager/v1/relevancy_manager.go:153]: Remove

File: pkg/networkmanager/network_manager.go
Comment (dwertent) [inline pkg/networkmanager/network_manager.go:164]: Change log level

File: pkg/networkmanager/network_manager.go
Comment (dwertent) [inline pkg/networkmanager/network_manager.go:170]: Change log level

File: pkg/networkmanager/network_manager.go
Comment (dwertent) [inline pkg/networkmanager/network_manager.go:178]: Change log level

File: pkg/networkmanager/network_manager.go
Comment (dwertent) [inline pkg/networkmanager/network_manager.go:184]: Change log level

**Review by dwertent — CHANGES_REQUESTED** [review]

**Review by dwertent — APPROVED** [review]

**Pull request merged**
